apps/web/src/components/Navigation/problem-explorer-track-nav.tsx

```diff
@@ -50,11 +52,23 @@ export function ProblemExplorerTrackNav({ isCollapsed, className }: ProblemExplo
         className,
       )}
     >
+      <Button
+        variant="ghost"
+        size="sm"
+        onClick={() => router.back()}
```

Comment:
I think using .back() might cause some issues in certain scenarios:

e.g. 
- if I enter the challenge directly from a URL, .back() won't do anything and the button is non-functional.
- if I enter the challenge from any other page than explorer page, it will take me back there (do we want that?)

I think we can solve this in two different ways:
- Always show back button and always redirect to `/explore`
- Only show back button if you can do .back(), hide otherwise

cc: @bautistaaa / @Hacksore thoughts?

Reply:
1. I think a back button should naturally lead the user back to where they came from. Hardcoding `/explore` doesn't feel like a necessary default.
2. We can have `/explore` as a default if the user landed directly on the challenge page (basically in cases where `router.back()` would be non-functional)

Reply:
Good point, but honestly not sure why we need this button if it's just going to behave exactly like the browsers built-in functionality?? @bautistaaa